### PR TITLE
Fix argument passing

### PR DESCRIPTION
**Pull request by zherczeg**
Unfortunately there was an incorrect subtraction in the original patch. Buffer is only updated if the execution is successful.

A Clang compiled coremark now runs with Walrus. Release, x86 results:

Interp:
```
2K performance run parameters for coremark.
CoreMark Size    : 666
Total ticks      : 464668
Total time (secs): 464.668000
Iterations/Sec   : 860.829668
Iterations       : 400000
Compiler version : Clang 18.1.2-wasi-sdk (https://github.com/llvm/llvm-project 26a1d6601d727a96f4301d0d8647b5a42760ae0c)
Compiler flags   : -O3 -DPERFORMANCE_RUN=1
Memory location  : Please put data memory location here
                        (e.g. code in flash, data on heap etc)
seedcrc          : 0xe9f5
[0]crclist       : 0xe714
[0]crcmatrix     : 0x1fd7
[0]crcstate      : 0x8e3a
[0]crcfinal      : 0x65c5
Correct operation validated. See README.md for run and reporting rules.
CoreMark 1.0 : 860.829668 / Clang 18.1.2-wasi-sdk (https://github.com/llvm/llvm-project 26a1d6601d727a96f4301d0d8647b5a42760ae0c) -O3 -DPERFORMANCE_RUN=1 / Heap
```

JIT:
```
2K performance run parameters for coremark.
CoreMark Size    : 666
Total ticks      : 57659
Total time (secs): 57.659000
Iterations/Sec   : 6937.338491
Iterations       : 400000
Compiler version : Clang 18.1.2-wasi-sdk (https://github.com/llvm/llvm-project 26a1d6601d727a96f4301d0d8647b5a42760ae0c)
Compiler flags   : -O3 -DPERFORMANCE_RUN=1
Memory location  : Please put data memory location here
                        (e.g. code in flash, data on heap etc)
seedcrc          : 0xe9f5
[0]crclist       : 0xe714
[0]crcmatrix     : 0x1fd7
[0]crcstate      : 0x8e3a
[0]crcfinal      : 0x65c5
Correct operation validated. See README.md for run and reporting rules.
CoreMark 1.0 : 6937.338491 / Clang 18.1.2-wasi-sdk (https://github.com/llvm/llvm-project 26a1d6601d727a96f4301d0d8647b5a42760ae0c) -O3 -DPERFORMANCE_RUN=1 / Heap
```

It looks like it validates its result, which is very good for correctness testing!

### CODE REVIEW

File: src/wasi/WASI.cpp [hunk 1]
```diff
@@ -118,17 +118,17 @@ void WASI::args_get(ExecutionState& state, Value* argv, Value* result, Instance*
 
     TemporaryData<void*, 8> pointers(argc);
 
-    if (pointers.data() == nullptr) {
```
Comment (zherczeg) [inline src/wasi/WASI.cpp:121]: These checks are unnecessary, because uvwasi checks its input pointers. Instead, we check that the result is success.

**Review by clover2123 — APPROVED** [review]
LGTM

**Comment by clover2123** [comment]
BTW Are you going to add coremark benchmark?

**Comment by zherczeg** [comment]
It is not a fast benchmark, but it is popular. I am not sure what is the best decision. Maybe it can be part of it, but not always run. What do you think?

**Comment by clover2123** [comment]
It seems that the total time taken for this benchmark is not that substantial.
What about adding this benchmark to our CI?

```
Interp Total time (secs): 464.668000
JIT Total time (secs): 57.659000
```

**Pull request merged**